### PR TITLE
Remove all use of deprecated PostgresNIO APIs

### PR DESCRIPTION
**Pull request by gwynne**
This is a major cleanup pass that deprecates `PostgresDataEncoder`, `PostgresDataDecoder`, and `PostgresConfiguration` in favor of working with the newer, better-designed APIs now available from PostgresNIO. As a side effect of this work, the minimum Swift version has been bumped to 5.7.

Every effort has been made to maintain backwards-compatible behavior whenever possible.

### CODE REVIEW

**Review by 0xTim — APPROVED** [review]

File: Tests/PostgresKitTests/PostgresKitTests.swift [hunk 1]
```diff
@@ -15,7 +15,7 @@ class PostgresKitTests: XCTestCase {
     }
     
     func testPerformance() throws {
-        let db = PostgresConnectionSource(configuration: .test)
+        let db = PostgresConnectionSource(sqlConfiguration: .test)
```
Comment (0xTim) [inline Tests/PostgresKitTests/PostgresKitTests.swift:18]: Do we have any deprecated APIs for the old one? I couldn't spot them

**Review by 0xTim — APPROVED** [review]
Looks good, want to double check we have deprecations for all the APIs that have been replaced

File: Tests/PostgresKitTests/PostgresKitTests.swift (hunk 1 above)
Comment (gwynne) [reply Tests/PostgresKitTests/PostgresKitTests.swift:18]: Yep, it's right here: https://github.com/vapor/postgres-kit/pull/241/files/995aa613ff3a3fb9196d666f31e2bdee3ad0adc3#diff-5c462d96b4d820bf87f30fef51a05da029789e0fb4359d1a9fed1793518b3dc4R9

File: Sources/PostgresKit/PostgresDataTranslation.swift [hunk 2]
```diff
@@ -0,0 +1,365 @@
...
+            }
+            bindings.append(legacyData)
+        }
+        /// Slow path: Descend through the ``Encodable`` machinery until we fail or find something we can convert.
+        else {
+            try bindings.append(self.encode(codingPath: [], userInfo: [:], value: value, in: context, file: file, line: line))
+        }
+    }
+    
+    internal /*fileprivate*/ static func encode<T: Encodable, E: PostgresJSONEncoder>(
```
Comment (0xTim) [inline Sources/PostgresKit/PostgresDataTranslation.swift:111]: Can we remove the file private from here?

File: Sources/PostgresKit/PostgresDataTranslation.swift [hunk 3]
```diff
@@ -0,0 +1,365 @@
...
+        func nestedUnkeyedContainer(forKey: K) -> any UnkeyedEncodingContainer { self }
+        func superEncoder() -> any Encoder { self }
+        func superEncoder(forKey: K) -> any Encoder { self }
+        func container<N: CodingKey>(keyedBy: N.Type) -> KeyedEncodingContainer<N> { .init(FailureEncoder<N>()) }
+        func unkeyedContainer() -> any UnkeyedEncodingContainer { self }
+        func singleValueContainer() -> any SingleValueEncodingContainer { self }
+    }
+
+    // TODO: Finish all this and make it work
+    /*
```
Comment (0xTim) [inline Sources/PostgresKit/PostgresDataTranslation.swift:335]: I'm assuming this is all still todo?

**Review by 0xTim — APPROVED** [review]
Couple of comments and lots of TODOs left over for referencd

File: Sources/PostgresKit/PostgresDataTranslation.swift (hunk 2 above)
Comment (gwynne) [reply Sources/PostgresKit/PostgresDataTranslation.swift:111]: This comment follows the same style used by NIO, used to indicate "this _should_ be `[file]private` but is `internal` so unit tests can access it.

File: Sources/PostgresKit/PostgresDataTranslation.swift (hunk 3 above)
Comment (gwynne) [reply Sources/PostgresKit/PostgresDataTranslation.swift:335]: Yeah, I should yank that stuff, even those bits would have to be redone anyway.

**Pull request merged**